### PR TITLE
`<DropdownLayout/>` & `<InputWithOptions/>` - add a11y roles

### PR DESCRIPTION
**Pull request by hepiyellow**
Just for fun. (And to learn a little)

Was using this page mainly for reference:
https://www.w3.org/TR/2010/WD-wai-aria-20100916/roles

### CODE REVIEW

File: src/DropdownLayout/DropdownLayout.js [hunk 1]
```diff
@@ -13,6 +13,11 @@ const modulu = (n, m) => {
 const NOT_HOVERED_INDEX = -1;
 export const DIVIDER_OPTION_VALUE = '-';
 
+const itemRoleByContainerRole = {
```
Comment (yardnsm) [inline src/DropdownLayout/DropdownLayout.js]: To make the intention more clear, I would use this as a function. For example:

```js
const mapItemRoleByContainerRole = role => {
	const rolesMap = {
		list: 'listitem',
		listbox: 'option',
	};

	return rolesMap[role];
};
```

File: src/DropdownLayout/DropdownLayout.spec.js [hunk 2]
```diff
@@ -609,6 +610,44 @@ describe('DropdownLayout', () => {
...
+      const driver = createDriver(
+        <DropdownLayout options={options} role="listbox" />,
+      );
+      expect(driver.element.getAttribute('role')).toBe('listbox');
+    });
+
+    it(`should have role 'listitem' for all non-divider options`, () => {
+      const driver = createDriver(<DropdownLayout visible options={options} />);
+      options.forEach((option, index) => {
+        if (option.value === DIVIDER_OPTION_VALUE) {
```
Comment (yardnsm) [inline src/DropdownLayout/DropdownLayout.spec.js]: You can use the `isOptionADivider` method of the driver instead

File: src/DropdownLayout/DropdownLayout.spec.js [hunk 3]
```diff
@@ -609,6 +610,44 @@ describe('DropdownLayout', () => {
...
+          expect(driver.optionAt(index).getAttribute('role')).toBe('separator');
+        } else {
+          expect(driver.optionAt(index).getAttribute('role')).toBe('listitem');
+        }
+      });
+    });
+
+    it(`should have role 'option' for all options and 'separator' for dividers`, () => {
```
Comment (yardnsm) [inline src/DropdownLayout/DropdownLayout.spec.js]: Maybe:

```suggestion
     it(`should have role 'option' for all options and 'separator' for dividers when role="listbox"`, () => {
```

File: src/InputWithOptions/InputWithOptions.private.driver.js [hunk 4]
```diff
@@ -0,0 +1,14 @@
+import inputWithOptionsDriverFactory from './InputWithOptions.driver';
+
+const inputWithOptionsPrivateDriverFactory = ({ element, wrapper }) => {
```
Comment (yardnsm) [inline src/InputWithOptions/InputWithOptions.private.driver.js]: Some for the private `<DropdownLayout/>` driver, all these methods are available in the public driver.

I noticed you're accessing `element` directly in order to use `getAttribute('role')`. Maybe you can add a `getRole()` method?

And as for the `dropdownLayoutElement`, you can access to `<DropdownLayout/>` driver in the `<InputWithOptions/>` driver:

https://github.com/wix/wix-style-react/blob/4573d14b88/src/InputWithOptions/InputWithOptions.driver.js#L75

File: src/DropdownLayout/DropdownLayout.spec.js (hunk 2 above)
Comment (hepiyellow) [reply src/DropdownLayout/DropdownLayout.spec.js]: will do

File: src/DropdownLayout/DropdownLayout.spec.js (hunk 3 above)
Comment (hepiyellow) [reply src/DropdownLayout/DropdownLayout.spec.js]: ok

File: src/DropdownLayout/DropdownLayout.js (hunk 1 above)
Comment (yardnsm) [reply src/DropdownLayout/DropdownLayout.js]: You can still leave it as an object, it's just my preference. And yeah, call it what you want :)

File: src/InputWithOptions/InputWithOptions.private.driver.js (hunk 4 above)
Comment (hepiyellow) [reply src/InputWithOptions/InputWithOptions.private.driver.js]: done

**Comment by hepiyellow** [comment]
Closing because of nested private driver hell !
Will resume later.

**Pull request closed without merging**